### DISCUSSION TITLE
Using VLW  fonts from Processing

### DISCUSSION THREAD

**Discussion by DamirAS2020** [Bodmer/TFT_eSPI#920] [Q&A - Fonts] [answered]
I need cyrillic fonts to use in a project.
But I do not know how to use VLW file in my sketch.
Please explain.
Thanks!

**Comment by Bodmer** [comment]
Have you created the vlw file using the provided Processing sketch?

**Reply by DamirAS2020** [reply]
Thanks for soon reply.

(I use Russian version of Processing. So tab names may differ.)

"File". "Examples": I choose some example.
"Tool". "Create font". Choose "Arial-BoldMT". Size "48". "Anti-aliasing."
"Symbols"   "Defined Unicode blocks". Choose "Cyrillic". "Create".

"Sketch". "Show sketches folder."
In folder Data I find vlw file.

**Comment by Bodmer** [comment, marked as answer]
What you do next depends on the processor you are using, e.g.  SPIFFS, LittleFS or use a flash array. See the library [Smooth Font examples](https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Smooth%20Fonts) and read the comments for more information. I suggest the arrray method unless you are already familiar with SPIFFS and LittleFS.

To convert the font to an array [see this example font.](https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Smooth%20Fonts/FLASH_Array/Font_Demo_1_Array/NotoSansBold15.h)

**Reply by DamirAS2020** [reply]
Done! Working!
Thanks a lot!
Trying both flash array and SPIFFS method.
But SPIFFS started to work only when I put "true" between () in SPIFFS. begin(). Also your examples did not work befour changing. (???)
Now it is possible to use. Bur cyrillic letters look not nice.
I tried to convert ttf to HEX. But  I 
![IMG_6827](https://user-images.githubusercontent.com/76564555/104087730-3b0b8d80-527b-11eb-81d8-69171bb72baf.jpg)
do not understand, how to control sizes.

**Comment by Bodmer** [comment]
SPIFFS. begin(true) will reformat the file system. The IDE will do this on upload of the file so changes to the sketches are not needed. Reformatting every time a sketch is run will take a lot of time. Read the SPIFFS documentation for further information.

It looks like you are not using the processing sketch provided with the library, that makes it easier. If using the Processing IDE menu to generate fonts the make sure you select the "smooth font" option.

**Reply by DamirAS2020** [reply]
![IMG_6830](https://user-images.githubusercontent.com/76564555/104105752-beec6680-52c9-11eb-9eb5-b4c1e74db1b6.jpg)
I tried to choose both cyrillic and basic latin together. Smooth flag was ok. Bad result again.

**Comment by Bodmer** [comment]
Compare your sketch to the working Smooth Font examples. Build your font into one of those working examplkes (with minimal other changes) to test

**Reply by DamirAS2020** [reply]
After spending a few hours, I found out what the problem was. 
Following your recommendations, the example from the library was slightly modified. 
The result exceeded expectations. The symbols looked beautiful!
Comparing this sketch with my own, I consistently commented out the problematic parts of the sketch: EEPROM, hardware timer, multitasking, DMA, external interrupts, etc.
Only then did I see the difference with the modified library example.
In the example "TFT. setTextColor (TFT_WHITE, TFT_BLACK);" were used two parameters.
But I used only one, as I was superimposing text on the picture. (Photo above)
If you set different colors for the screen background and text background, you can see the anti-aliasing dots on the screen as a result.

  tft.fillScreen(TFT_GREEN);
  tft.setTextColor(TFT_WHITE, TFT_BLACK);
  tft.print("TFT_eSPI is the best!");

So, if I understand correctly, it is impossible to put on smoothed characters above the image.


https://user-images.githubusercontent.com/76564555/104117242-2a215180-5339-11eb-8afa-84956c343d41.mov

**Comment by Bodmer** [comment]
If you wish to put anti aliased font over a screen image then the screen RAM must be read to get the background colour. See "Smooth_font_reading_TFT".

**Comment by DamirAS2020** [comment]
Done! Tested!
Fantastic library for all cases of the life!
Nice work!
Thanks! Thanks! Thanks!
